src/lpython/semantics/python_ast_to_asr.cpp

```diff
@@ -1914,6 +1914,65 @@ class BodyVisitor : public CommonVisitor<BodyVisitor> {
             }
             tmp = ASR::make_ConstantLogical_t(al, x.base.base.loc, result, type);
             return;
+        } else if (call_name == "int") {
+            if (args.size() != 1) {
+                throw SemanticError(call_name + "() takes exactly one argument (" +
+                    std::to_string(args.size()) + " given)", x.base.base.loc);
+            }
+            ASR::expr_t* int_expr = args[0];
+            ASR::ttype_t* int_type = ASRUtils::expr_type(int_expr);
```

Comment:
Let's check that `ASRUtils::expr_value(int_expr)` is not null here. If it is null, give a compile time error that runtime `int(x)` is not supported, only compile time for now.